sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -169,7 +173,8 @@ az_build_headers(az_http_request * p_request, struct curl_slist ** p_headers) {
  * @param url_from_request a url that is not zero terminated
  * @return az_result
  */
-AZ_NODISCARD az_result az_write_url(az_span writable_buffer, az_span url_from_request) {
+static AZ_NODISCARD az_result
+_az_http_client_curl_write_url(az_span writable_buffer, az_span url_from_request) {
 
   AZ_RETURN_IF_FAILED(az_span_append(writable_buffer, url_from_request, &writable_buffer));
   AZ_RETURN_IF_FAILED(az_span_append(writable_buffer, AZ_SPAN_ZEROBYTE, &writable_buffer));
```

Comment:
Is zero-byte right? I think it is CRLF again

Reply:
At some point, we should ultimately add '\0', because curl expects 'char *' strings to be null-terminated. But as far as header separator, I think you're right, there's a bug and this is likely why I currently can't set Content-Type header for the token request.

Reply:
do curl's doc say this explicitly? Maybe curl wants some length which doesn't require 0-termination. We should know; not guess.